### PR TITLE
Feature/registration

### PR DESCRIPTION
**Pull request by sharansp**

### CODE REVIEW

File: src/app/register/register.component.spec.ts [hunk 1]
```diff
@@ -1,25 +1,99 @@
...
+    component.registerFormGroup.controls['name'].setValue("abc1")
+    expect(component.registerFormGroup.controls['name'].hasError('pattern')).toBeTruthy()
+  }));
+
+  it('Name should contain min 3 characters', async(() => {
+    component.registerFormGroup.controls['name'].setValue("ab")
+    expect(component.registerFormGroup.controls['name'].hasError('minlength')).toBeTruthy()
+  }));
+
+  it('Name should not contain more than 20 characters', async(() => {
```
Comment (reddy-s) [inline src/app/register/register.component.spec.ts]: @sharansp : I think name can be more than 20 characters mate! May be you can make it 60 characters

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -15,7 +15,8 @@ const routes: Routes = [
 ];
 
 @NgModule({
-  imports: [ RouterModule.forRoot(routes) ],
+  imports: [ RouterModule.forRoot(routes)
```
Comment (reddy-s) [inline src/app/app-routing.module.ts]: i don't think any change is needed here

File: src/app/register/register.component.ts [hunk 3]
```diff
@@ -1,52 +1,171 @@
...
-  imports: [
-    FormBuilder,
-    Validators,
-    FormGroup,
-    MatStepperModule,
-    MatInputModule,
-    MatButtonModule,
-    MatAutocompleteModule
-  ]
+	imports: [
```
Comment (reddy-s) [inline src/app/register/register.component.ts]: indentation is more than needed here!

File: src/app/register/register.component.ts [hunk 4]
```diff
@@ -1,52 +1,171 @@
...
+	private _resendConfirmationCodeService: ResendConfirmationCodeService
+	private _signUpService: SignUpService
+	_registration: Registration = new Registration();
+
+	// response data/flag to show/handle in UI
+	registrationConfirmed: Boolean = false;
+	_resentConfirmationCodeResponse: Boolean = false;
+
+	registerFormGroup: FormGroup // UI reactive Form Group variable 
+	mobilePrefix = "+91"
```
Comment (reddy-s) [inline src/app/register/register.component.ts]: Why do we need this?

File: src/app/register/register.component.ts [hunk 5]
```diff
@@ -1,52 +1,171 @@
...
+		})
+	}
+
+  // Onclick register/submit of first stepper section
+	registerUser(formData) {
+		this.resetResponseMessages()
+    this.progressSpinner = true
+    // mapping input values
+		this._registration = new Registration(
+			['+91' , formData.mobileNumber].join(""),
```
Comment (reddy-s) [inline src/app/register/register.component.ts]: here as well! I think you should start using constant which @AnirupPat created. It should be in the utils/constants.ts folder

Comment (sharansp) [reply src/app/register/register.component.ts]: done

File: src/app/register/register.component.ts (hunk 4 above)
Comment (sharansp) [reply src/app/register/register.component.ts]: updated

File: src/app/register/register.component.spec.ts (hunk 1 above)
Comment (sharansp) [reply src/app/register/register.component.spec.ts]: Changed to 60

File: src/app/register/register.component.ts (hunk 3 above)
Comment (sharansp) [reply src/app/register/register.component.ts]: it's same throughout

File: src/app/app-routing.module.ts (hunk 2 above)
Comment (sharansp) [reply src/app/app-routing.module.ts]: reverted

**Pull request merged**